test/document.test.js

```diff
@@ -14296,6 +14296,93 @@ describe('document', function() {
 
     delete mongoose.Schema.Types.CustomType;
   });
+
+  it('supports schemaFieldsOnly option for toObject() (gh-15258)', async function() {
+    const schema = new Schema({ key: String }, { discriminatorKey: 'key' });
+    const subschema1 = new Schema({ field1: String });
+    const subschema2 = new Schema({ field2: String });
+
+    const Discriminator = db.model('Test', schema);
+    Discriminator.discriminator('type1', subschema1);
+    Discriminator.discriminator('type2', subschema2);
+
+    const type1Key = 'type1';
+    const type2Key = 'type2';
+
+    const doc = await Discriminator.create({
+      key: type1Key,
+      field1: 'test value'
+    });
+
+    await Discriminator.updateOne(
+      { _id: doc._id },
+      {
+        key: type2Key,
+        field2: 'test2'
+      },
+      { overwriteDiscriminatorKey: true }
+    );
+
+    const doc2 = await Discriminator.findById(doc).orFail();
+    assert.strictEqual(doc2.field1, undefined);
+    assert.strictEqual(doc2.field2, 'test2');
```

Comment:
```suggestion
    assert.strictEqual(doc2.field2, 'test2');
    assert.strictEqual(doc2.field1, undefined);
```

Align with the style of the following 2 checks